### PR TITLE
Issue #21, Issue #22

### LINKED ISSUE #21: [Student A] Liczba zdań w pliku

### LINKED ISSUE #22: [Student B] Zatroszczyć się o poprawność raportu po wszystkichzmianach.
Rozszerzyć raport o ilość zdań i wyrazów. Osobno wyświetlić samogłoski i spółgłoski.

### PR DESCRIPTION
**Pull request by MichalRac**
fixes #21 - Leaving only '?' and ',' on the list of end characters.
fixes #22 - some general flow adjustments and fixes where neccessary.

### CODE REVIEW

**Review by piotrkardasz — APPROVED** [review]
Tested 🧐 work's correctly

File: TextAnalyzer/Main.cs [hunk 1]
```diff
@@ -22,37 +22,53 @@ public Main()
...
 
+            List<ICommand> options = new List<ICommand>{ optionsList[downloadIndex], optionsList[exitIndex] };
+            return options;
+        }
+
+        private void WriteOptionsAndTakeChoice(List<ICommand> viableOptions)
+        {
             int i = 1;
-            foreach (var command in optionsList) writeMessege($"{i++}. {command.Description}");
+            foreach (var command in viableOptions) writeMessege($"{i++}. {command.Description}");
```
Comment (piotrkardasz) [inline TextAnalyzer/Main.cs]: I suggest to use more strict variables and more readable form
```suggestion
            foreach (ICommandin viableOptions)
            {
                    writeMessege($"{i++}. {command.Description}"
            );
```

Comment (MichalRac) [reply TextAnalyzer/Main.cs]: I agree, especially about the form, though it was switched from a this type of convention to the one used before in one of the early PR as part of the review.
Also, i think you are missing an identifier and a space in this foreach loop, so I suggest something like this:
```suggestion
            foreach (ICommand command in viableOptions)
            {
                    writeMessege($"{i++}. {command.Description}"
            );
```

Comment (MichalRac) [reply TextAnalyzer/Main.cs]: It also seems that there was a missing semicolon, we have to be carefull with suggstions via github as they bring potential risks.

**Comment by MichalRac** [comment]
Adjusted to feedback, squashed and merged in.

**Pull request merged**